webapp/channels/src/components/preparing_workspace/roles.tsx

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect} from 'react';
+import {FormattedMessage, useIntl} from 'react-intl';
+import {CSSTransition} from 'react-transition-group';
+
+import {Animations, mapAnimationReasonToClass, Form, PreparingWorkspacePageProps} from './steps';
+
+import Title from './title';
+import Description from './description';
+import PageBody from './page_body';
+import SingleColumnLayout from './single_column_layout';
+
+import PageLine from './page_line';
+import './roles.scss';
+import {useDispatch, useSelector} from 'react-redux';
+import {getWorkTemplateCategories as fetchCategories} from 'mattermost-redux/actions/work_templates';
+import {getWorkTemplateCategories} from 'selectors/work_template';
+import classNames from 'classnames';
+import Input from 'components/widgets/inputs/input/input';
+import {CategoryOther} from '@mattermost/types/work_templates';
+
+type Props = PreparingWorkspacePageProps & {
+    role: Form['role'];
+    roleOther: Form['roleOther'];
+    setRole: (role: string, roleOther: string) => void;
+    className?: string;
+    quickNext(role: string): void;
+}
+const Roles = ({role, next, ...props}: Props) => {
+    const {formatMessage} = useIntl();
+    const dispatch = useDispatch();
+    const categories = useSelector(getWorkTemplateCategories);
+
+    let className = 'Roles-body';
+
+    useEffect(() => {
+        dispatch(fetchCategories());
+    }, [dispatch]);
```

Comment:
nit: Guessing that you want categories to be fetched once after mount, maybe adding `dispatch` as dep could cause it to be fetched multiple times, and causing re-renders.